src/components/dialogs/create-case-dialog.tsx

```diff
@@ -197,53 +198,39 @@ const CreateCaseDialog: React.FunctionComponent<ICreateCaseDialogProps> = ({
     ]);
 
     return (
-        <FormProvider {...createCaseFormMethods}>
-            <Dialog
-                fullWidth={true}
-                open={open}
-                onClose={handleCloseDialog}
-                aria-labelledby="create-case-form-dialog-title"
-            >
-                <DialogTitle id="create-case-form-dialog-title">
-                    <FormattedMessage id="ImportNewCase" />
-                </DialogTitle>
-                <DialogContent>
-                    <TextFieldInput
-                        label={'nameProperty'}
-                        value={caseName}
-                        setValue={(newValue) => setValue(CASE_NAME, newValue)}
-                        error={caseNameErrorMessage}
-                        autoFocus
-                        adornment={caseFileAdornment}
-                        setValueHasChanged={setCaseNameChanged}
-                    />
-                    <TextFieldInput
-                        label={'descriptionProperty'}
-                        value={description}
-                        setValue={(newValue) => setValue(DESCRIPTION, newValue)}
-                    />
-                    {caseFileErrorMessage && (
-                        <Alert severity="error">{caseFileErrorMessage}</Alert>
-                    )}
-                    <UploadNewCase
-                        caseFile={caseFile}
-                        handleCaseFileUpload={handleCaseFileUpload}
-                    />
-                </DialogContent>
-                <DialogActions>
-                    <Button onClick={handleCloseDialog} variant="text">
-                        <FormattedMessage id="cancel" />
-                    </Button>
-                    <Button
-                        onClick={handleSubmit(handleCreateNewCase)}
-                        disabled={!isCreationAllowed}
-                        variant="outlined"
-                    >
-                        <FormattedMessage id="validate" />
-                    </Button>
-                </DialogActions>
-            </Dialog>
-        </FormProvider>
+        // @ts-ignore
+        <CustomMuiDialog
+            titleId={'ImportNewCase'}
+            formSchema={schema}
+            formMethods={createCaseFormMethods}
+            removeOptional={true}
+            open={open}
+            onClose={handleCloseDialog}
+            onSave={handleCreateNewCase}
+            disabledSave={!isCreationAllowed}
+        >
+            <TextInput
```

Comment:
You need to create a new component especially for `name` that will do all the checks we need and then delete useNameCheck.

Reply:
I forgot we created a TS for this special case, so you don't have to do this in this PR

Reply:
ok, thank you